tests/msc2946_test.go

```diff
@@ -207,9 +208,16 @@ func TestClientSpacesSummary(t *testing.T) {
 		// should omit either R3 or R4 if we start from SS1 because we only return 1 link per room which will be:
 		// SS1 -> SS2
 		// SS2 -> R3,R4 (but only 1 is allowed)
-		res := alice.MustDo(t, "POST", []string{"_matrix", "client", "unstable", "org.matrix.msc2946", "rooms", ss1, "spaces"}, map[string]interface{}{
-			"max_rooms_per_space": 1,
-		})
+		query := make(url.Values, 1)
+		query.Set("max_rooms_per_space", "1")
+		res := alice.MustDoRaw(
```

Comment:
why does this one need a `MustDoRaw` ?

Reply:
`MustDo` doesn't seem to accept query parameters.

Reply:
oh yes. That's odd, given `Do` does. fair enough.

Reply:
I can attempt to pipe it all through if you'd prefer, not sure if we were trying to keep those APIs extremely simple or not (maybe @Kegsay has a thought)?

Reply:
Having seen more people use the client, I would press for functional options at this point. That would avoid the API mismatches that we have here between `MustDo` and `Do`, and make it clearer on the caller side what checks you're doing e.g:
```go
alice.MustDo(t, "POST", []string{"_matrix", "client", "unstable", "org.matrix.msc2946", "rooms", ss1, "spaces"}, map[string]interface{}{
	"max_rooms_per_space": 1,
}, client.WithQueries(query), client.WithReturnStatusCode(200))
```
etc. Exactly what shape this will take I don't know yet - some are assertions and some modiy the request. Should they all be functional options entirely or will that get annoying doing things like:
```go
alice.Do(t,
	client.WithMethod("POST"),
	client.WithPath([]string{"_matrix", "client", "unstable", "org.matrix.msc2946", "rooms", ss1, "spaces"}),
	client.WithBody(map[string]interface{}{
		"max_rooms_per_space": 1,
	}),
	client.WithQueries(query),
	client.WithAuth(),
)
```
This would also make it clearer for cases where you aren't sending a body (as you don't need to awkwardly set `nil` as the arg, you just omit `client.WithBody(..)`. The downside is that the API then isn't guiding the caller into what bits are required vs optional, so maybe a hybrid where method/path are required, but then it's all optional functions?